### ISSUE TITLE
Copay Watch Only Wallet with Trezor - No Longer Functions

### ISSUE THREAD

**Issue by jwebbstevens** [bitpay/copay#6960]
Pre-Segwit one used to be able to setup Copay to be a watch only wallet.

This was extremely useful for wallet users who have funds in a [Trezor](https://trezor.io/) to be able to access details safely from another secure source like Copay to monitor whats in the hardware wallet.

[Creating watch-only wallet on any mobile device](https://doc.satoshilabs.com/trezor-apps/copay.html#creating-watch-only-wallet-on-any-mobile-device)

With the release of Segwit and Trezor's migration and support of it they separated pre-segwit accounts into what they call "Legacy" accounts.  Those account had XPUBs which, to my understanding, was a key component for the interaction between Copay as a watch only wallet and a Trezor hardware wallet. 

Now what happens is that when one attempts to set this up via Trezors (older) instructions is that Copay just hangs/times out and you can't setup the watch only wallet anymore.  

I think this is both a bug and a feature since Copay doesn't support Segwit quite yet and the old methodology no longer works.  It would be swell if this could work again.

**Comment by bradphilips** [comment]
This is happening to me too. Any word on when this will be supported?

**Comment by bradphilips** [comment]
As a side note I accidentally sent funds thinking it would go to the segwit account but it went to the automatically created legacy account. I had to transfer it back incurring fees.

**Comment by matiu** [comment]
Please talk to Trezor team regarding this issue, we are not aware of the changes they made to their wallet or the Watch-only procedure you are mentioning. I am sorry we can't help you with this one.

**Comment by jwebbstevens** [comment]
hey @bradphilips your second comment has to do with trezor and how they segregated Segwit and non-segwit wallets in advance of the chain split.  You can chat with them directly about that over at [https://trezor.io/support/](https://trezor.io/support/).  They have good documentation about legacy accounts on the site as well as other channels of communication like reddit, etc.

@matiu I think I did a poor job of explaining what is happening and which part is related to Copay and what is related to Trezor teams. I'll put something together with screenshots in attempt to explain what's happening more clearly.

**Comment by matiu** [comment]
@jwebbstevens thanks!

**Comment by jwebbstevens** [comment]
So here's some more/better detail (maybe - fingers crossed)

Prior to the impending chain split as well as after some updates to the Copay Mobile Wallet users used to be able to setup Copay as a watch only wallet.

This would take advantage of some features and elements of bitcoin wallets that are rapidly changing now.  Specifically, here is what, according to Trezor a user would have to do.

![watch only wallet instructions for copay wallet](https://user-images.githubusercontent.com/7277574/32245810-5a6bf990-be53-11e7-845a-9254309186f0.png)

Copay's interface has changed a little and Trezor has changed something about how XPUBs are used/available. According to Trezor Team the XPUBS are available only for Legacy accounts (pre-segwit).

It used to be that you could use the Copay Chrome App desktop client to do this but because Trezor doesn't have those XPUBS anymore/temporarily because no 3rd party wallets are supporting SegWit yet and that Copay has made changes as it relates to wallet imports with updates to the app, users can't take advantage of setting up a watch-only wallet.

So my understanding is nothing can happen other than notifying the dev teams of each respective application/tool that:

1. Copay has to support SegWit in order for a watch only wallet to work
2. Trezor updates [setup steps](https://doc.satoshilabs.com/trezor-apps/copay.html#creating-watch-only-wallet-on-any-mobile-device) and makes XPUBs available for interaction on the part of the user and other wallets/services.
3. Copay needs to consider updating its Chrome App regarding this as well.

I'm going to link this github issue back to the support Trezor person who let me know why Trezor isn't working. Maybe he/she can get some of the Trezor team members to comment on the issue so they can chat with the Copay team on the issue.

If there is a way I could debug and provide whats going on to Copay when trying to run the Copay Chrome app for this I'm all ears.

**Comment by bradphilips** [comment]
@jwebbstevens, this is a wonderful description. Thank you, I hope this helps team. This is exactly what I've seen as well. I'm happy to try and help as well although I'd have to get myself familiar with the application. Without Trezor sharing the xpubs though, can Copay do anything until they support it?

**Comment by jwebbstevens** [comment]
@bradphilips sure thing.  I plan on taking screenshots step by step in the next few days so the Copay team can have more details. I've notified Trezor and they forwarded this to people on their dev team so hopefully that aids communication between the teams.

Copay can do nothing with out fixing the bugs on their side and also Copay needs to support SegWit so it will be a bit before this can be resolved or improved upon. I believe the segwit date was projected to be today (Nov 1, 2017).

**Comment by bradphilips** [comment]
@jwebbstevens, @matiu It looks like this is supported in [v4 of the trezorconnect](https://github.com/trezor/connect/tree/v4) lib, see [here](https://connect.trezor.io/4/examples/xpubkey.html).

This lists the new Segwit accounts but for some reason copay can't recover it from the XPUB that is  produced, not sure why this is yet but it looks like copay just needs to update the trezorconnect dependency to v4 to get started on the implementation of this:

```
"trezor-connect": "https://github.com/trezor/connect.git#541c57beac7e2707453ef416eb7fcceed1141981",
```

Here is a screenshot that shows it using account discovery:

<img width="712" alt="screen shot 2017-11-03 at 9 06 14 pm" src="https://user-images.githubusercontent.com/1027864/32400858-17f14a4e-c0dc-11e7-929e-7ad451fe1950.png">

**Comment by jwebbstevens** [comment]
Well now that SegWit2X is dead, Copay still needs to support plain old USAF segwit  from what I understand.  I reached out to the Trezor team again and they said they would monitor this issue along with others so I think this is DOA until Copay and Trezor want to work together to make this work again. I don't know what monitor means to them.  I asked if they would at least respond on this Copay Github issue to keep the conversation alive and they said they would try.

I can't even get the Trezor team to update the Help Docs to remove the feature that doesn't work.

https://doc.satoshilabs.com/trezor-apps/copay.html#creating-watch-only-wallet-on-any-mobile-device

Go figure.  

This is one of the major problems with Bitcoin and the companies around its development environment, the environment is too fluid for widespread adoption because "features" appear and are extremely useful, but then disappear and no one thinks to update there own docs to say that it is deprecated for the time being.

Bitcoin seems like a technology application with great promise but I can't imagine many will adopt Bitcoin in a meaningful way until functionality and stability increases.  Imagine if your run of the mill bank just stop providing you monthly statements and said "oh, its because XYZ company no longer supports ABC element so ¯\_(ツ)_/¯"  It's fundamental, the features have to be stable and widely adopted standards. Maybe one day this will get fixed.  Instead of just immediately closed.

**Comment by bradphilips** [comment]
@jwebbstevens, this is disappointing... I'm almost wondering if Copay just needs to update bitcore to support Segwit though and update to v4 of the trezorconnect SDK. I can at least get my Segwit xPub out of trezor, it just doens't satisfy Copay because it's not using the right derivation. I'll see what I can figure out with bitcore but I think you're right -- DOA until trezor and copay work together on this. 

Just FYI, I have found a wallet that works as watch only for Trezor at least on iOS called [bitWallet](http://www.sollico.com/bitwallet/) -- not sure which mobile device you prefer.

Go [here](https://connect.trezor.io/4/examples/xpubkey.html) and export the xPub for the Segwit account -- this is v4 trezor connect.

In bitWallet, go to Add a new wallet, specify HD Wallet -> Restore a Wallet -> Enter the xPub in the Extended Key or Mnemonic and choose Segwit on the account type.

Makes me sad I got a Trezor, should've went for the Ledger Nano... Hope this helps you at least. Thanks for all your help!

**Comment by jwebbstevens** [comment]
@bradphilips Yes, I think until Copay support Segwit, which my understanding is supporting the USAF Segwit (and not the Segwit 2X version which is DOA because the dev team dropped out), and then Trezor sees stability in the wallet space that this could be supported again.

It's still contingent on the Copay team refactoring this as well as the Trezor team. And since this issue thread was marked Closed I don't think it will be looked at.

I didn't know Ledger Nano and Copay worked together as a watch only. I'll have to check that out.

**Comment by bradphilips** [comment]
@jwebbstevens, @matiu - could this update help this issue?

https://blog.trezor.io/explore-the-new-send-dialogue-in-trezor-wallet-feae1253443a?source=linkShare-1778b3f3a3a-1513122022

**Comment by jwebbstevens** [comment]
Yes,  Copay, as I understand it still has to upgrade to support Segwit. But now that Trezor has opened up the ability to interact with XPUBS again it might work. The new segwit version of XPUBS I guess Trezor is calling YPUB now.

![screen shot 2017-12-13 at 10 26 36 am](https://user-images.githubusercontent.com/7277574/33946677-64e284ac-dff0-11e7-9d35-834014c5dcad.png)

Having a watch only wallet in Copay would be a really great feature. I've been following the Copay Segwit thread on another issue for a while now. It doesn't look like its been upgrade yet as they are focusing development on Bitcoin Cash and the UI.
